examples/bri-3/blockchain/blockchain.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+
+@Injectable()
+export class BlockchainService implements OnModuleInit {
+  async onModuleInit() {
+    // TODO: connect to blockchain
+  }
+
+  async deploy(contract: any) {
+    //TODO: deploy contract
+  }
+
+  async store(proof: string) {
+    //TODO: store content-addressable hash on contract
+  }
+
+  async verify(proof: string): Promise<boolean> {
+    //TODO: verify content-addressable hash exists in shield contract
```

Comment:
@biscuitdey is `verify` doing the verification offchain? or are we verifying onchain and by doing so finalizing the proof onchain? or will we do both? Let's decide cc @Kasshern @skosito @ognjenkurtic @Manik-Jain @Ybittan

Reply:
left similar comment, my vote is to do it the same as current shield contract in baseline repo for consistency, verifyAndPush method doing both, but no strong opinion on this

Reply:
@skosito the original architecture with shield contract etc was, to put it gently, highly questionable -> no L2 does zkp validation. Plus the nomenclature is awkward and misleading -- anyone should be able to verify a proof, that is stored/referenced onchain either onchain or offchain. I actually have quite a strong opinion about that

Reply:
I agree with @Therecanbeonlyone1969 
we shall only read the current state from BKC and run the validation off-chain.

Reply:
@Therecanbeonlyone1969 thanks for info, i was not around when shield contract was created and having it part of core package gave impression to me that this is accepted way of doing things
maybe good candidate for some BLIP after this project is done to fix this too since can create confusion

Reply:
Yes, the verification is happening on-chain. The current iteration will focus on state synchronization. Future iterations will add privacy using zero-knowledge circuits.
1. Participants will convert their documents (JSON) into a merkle tree. The merkle tree root will be the content-addressable hash of the document.
2. This hash (merkle tree root) will be stored in a smart contract (on-chain) in a map.
3. Alice will send her version of the document (DocA) to Bob. If Bob wants to verify DocA, he will turn it into a merkle tree, get the root and check if that root exists in the map that is stored in the smart contract.
4. Bob can update the DocA → DocB, and then repeat this process.